lir/src/number.rs

```diff
@@ -0,0 +1,156 @@
+/// This module implement all Number types, to specialize operations for a given
+/// numerical type, and to encode constant with the proper numerical type too.
+use std::hash::{Hash, Hasher};
+
+/// NumberType are used for math and bitwise operators. All other number types
+/// can be convert to this one, including NumberValue, using the `into()`
+/// method.
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq, Hash, Clone, Copy)]
+pub enum NumberType {
```

Comment:
Is it interesting to support `i128` and `u128` here?

Reply:
The problem with having `NumberType` for `i128` and `u128` is that we need a corresponding `NumberValue` and at the moment `NumberValue` are part of the `Opcode::Const`, Thus increasing the size of all opcodes.

Support for 128 bits number is something which would have to be handled to support Rust fully, but this can be expressed as 64 bits operations.

Reply:
Make sense! Thanks.